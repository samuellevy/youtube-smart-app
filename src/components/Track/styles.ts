import styled from 'styled-components';

export const Container = styled.div`
  overflow: hidden;
  height: 80vh;
`;
export const Content = styled.div`
  display: flex;
`;
export const CategoryTitle = styled.h2``;
