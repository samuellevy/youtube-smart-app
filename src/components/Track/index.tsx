import React, { useEffect, useState } from 'react';
import { useKeyboardContext } from '../../context/KeyboardContext';
import VideoCard from '../VideoCard';

import * as S from './styles';

interface ITrackComponent {
  active?:boolean;
  title?:string;
  id?:number;
}

interface IVideoCardItem {
  title: string;
  author: string;
  views: string;
  publishDate: string;
  active?: boolean;
}

const Track: React.FC<ITrackComponent> = ({ active, title, id }:ITrackComponent) => {
  const data = [
    {
      title: 'Harry Potter e a Pedra Filosofal',
      author: 'Warner Bros.',
      views: '1.3M',
      publishDate: '1 year ago',
    },
    {
      title: 'Harry Potter e a CÃ¢mara Secreta',
      author: 'Warner Bros.',
      views: '2.3M',
      publishDate: '2 years ago',
    },
  ];

  const [activeModule, setActiveModule] = useState(false);
  const [activeItem, setActiveItem] = useState(0);

  const { keyboard, dispatch }: any = useKeyboardContext();

  useEffect(() => {
    if (active === false) {
      setActiveItem(-1);
    } else {
      setActiveItem(0);
    }
  }, [active]);

  useEffect(() => {
    if (keyboard.component === 'home') {
      console.log(keyboard);
    }
  }, [keyboard]);

  return (
    <S.Container>
      <S.CategoryTitle>{title}</S.CategoryTitle>
      <S.Content>
        {data.map((item, key) => (
          <VideoCard data={item} active={activeItem === key} />
        ))}
      </S.Content>
    </S.Container>
  );
};

export default Track;
