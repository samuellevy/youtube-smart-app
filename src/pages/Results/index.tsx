/* eslint-disable array-callback-return */
import React, { useEffect, useState } from 'react';
import { useHistory } from 'react-router-dom';
import Track from '../../components/Track';
import { useFavoritesContext } from '../../context/FavoriteContext';
import { useKeyboardContext, changeComponent } from '../../context/KeyboardContext';
import { IVideo } from '../../dtos/IVideo';
import { IVideoCategory } from '../../dtos/IVideoCategory';
import { IVideoEntityReq } from '../../dtos/IVideoEntityReq';
import { api } from '../../services/mockapi';

import * as S from './styles';

const Results: React.FC = () => {
  const { keyboard, dispatch }: any = useKeyboardContext();
  const { favorites }: any = useFavoritesContext();
  const [activePage, setActivePage] = useState(false);
  const [activeItem, setActiveItem] = useState(0);
  const [trackLoading, setTrackLoading] = useState(true);
  const history = useHistory();

  const [resultData, setResultData] = useState<IVideo[]>([] as IVideo[]);

  const [videoCategories, setVideoCategories] = useState<IVideoCategory[]>([] as IVideoCategory[]);

  useEffect(() => {
    dispatch(changeComponent('results'));
    getCategories();
    // // getDataSearch();
    // setActiveItem(0);
    setActivePage(true);
  }, []);

  useEffect(() => {
    if (keyboard.component === 'menu') {
      setActivePage(false);
    }
  }, [keyboard]);

  const handleOut = () => {
    dispatch(changeComponent('menu'));
  };

  const getCategories = async () => {
    const tmpVideoCategories: IVideoCategory[] = [
      { id: 'query', title: 'Pesquisa', videos: [] }];

    setVideoCategories(tmpVideoCategories);
  };

  const handlePlay = (id:string) => {
    history.push(`/video/?${id}`);
  };

  return (
    <S.Container active={activePage} activeModule={activeItem > 0 ? activeItem : 0}>
      {videoCategories.map((item: IVideoCategory, key) => (
        <Track
          key={item.id}
          active={key === activeItem}
          title={item.title}
          id={item.id}
          item={item}
          handleOut={() => { handleOut(); }}
          handlePlay={(id: string) => { handlePlay(id); }}
          // directData={resultData}
          query="iza"
        />
      ))}
    </S.Container>
  );
};
export default Results;
