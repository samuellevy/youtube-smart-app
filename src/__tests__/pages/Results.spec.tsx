import React from 'react';
import Enzyme, { shallow, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import Results from '../../pages/Results';
import Track from '../../components/Track';

Enzyme.configure({ adapter: new Adapter() });

describe('<Results /> component', () => {
  it('should render properly', () => {
    const tree = renderer.create(<Results />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
