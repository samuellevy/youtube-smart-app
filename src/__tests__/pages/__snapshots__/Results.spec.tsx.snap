// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Results /> component should render properly 1`] = `
<div
  className="sc-jrAGrp fiSZBI"
/>
`;
